src/clojure/monger/util.clj

```diff
@@ -73,3 +73,8 @@
   (get-id
     [^IPersistentMap object]
     (or (:_id object) (object "_id"))))
+
+(defn into-array-list
+  "Coerce a j.u.Collection into a j.u.ArrayList."
+  ^java.util.ArrayList [^java.util.Collection coll]
```

Comment:
Interesting, the docs say [return value type hints can be placed before the argument vector](http://clojure.org/reference/java_interop#typehints).